bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonDecoder.kt

```diff
@@ -34,8 +34,18 @@ import org.bson.BsonReader
 import org.bson.BsonReaderMark
 import org.bson.BsonType
 import org.bson.BsonValue
+import org.bson.codecs.BooleanCodec
```

Comment:
I thought we discussed not using codecs (so to not confuse the two concepts of codecs and kserializers. Instead use the `NumberCodecHelper` directly (which is what the numeric codecs are doing in decode anyway).

Reply:
Thank you for the reminder. We did discuss prioritizing clarity by not mixing codecs with serializers. I've now implemented the changes to use the codec helpers directly as we discussed.